### PR TITLE
#171380743 Deploy this app to heroku

### PR DESCRIPTION
**Pull request by descholar-ceo**
#### What does this PR ​ do​ ?
It deploys this Application to heroku

#### Description of Task to be completed?
* Add *Procfile* file which contains the web process to start when this app is deployed to heroku
* Add *app.json* file to tell heroku how to create *REVIEW APPS* whenever a new PR is raised
* Add a sample route in *index.js* file to test it the deployment worked

#### How should this be manually tested?
* Click [here]() if you see in your browser this message *{"message":"welcome"}* then the deployment is successful or scroll dow to this PR's deployment history, and click on `view deployment`

#### Any background context you want to provide?
To know how to use this deployment please read this [README](https://github.com/Stackup-Rwanda/stackup2-barefoot-backend/blob/ch-deploy-base-app-to-heroku-171380743/README.md) file on *Deployments* section

#### What are the relevant pivotal tracker stories?
[#171380743](https://www.pivotaltracker.com/n/projects/2435482)
#### Screenshots (​ if​ appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by muchristian — APPROVED** [review]
Procfile and package.json files are requesting a newline at the end; actually you did a good job

**Review by muchristian — APPROVED** [review]
Procfile and package.json files are requesting a newline at the end please check it; actually you did a good job

**Comment by descholar-ceo** [comment]
@muchristian I resolved that issue, of newline

**Review by Muhire-Josue — COMMENTED** [review]
Good job @descholar-ceo, could you add a header on your commit message? ex: chore(deploy on Heroku).

**Comment by Muhire-Josue** [comment]
Great job

**Comment by descholar-ceo** [comment]
@Muhire-Josue The header is there 
Screenshot from 2020-03-01 21-18-20

[Here is the link to the commit message](https://github.com/Stackup-Rwanda/stackup2-barefoot-backend/pull/9/commits/bb514c74f9da4c731aedb64ff799d579b064a96d)

**Review by Muhire-Josue — APPROVED** [review]
Looks good to me.

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "test": "nyc mocha --require @babel/register ./tests/*.js  --timeout 25000 --exit",
+    "start": "set NODE_ENV=production && babel-node ./src/index.js",
```
Comment (ricahcyuzuzo) [inline package.json]: For the production, Do you need to use the same folder for development? Just use the babel generated files!

File: src/index.js [hunk 2]
```diff
@@ -1,93 +1,59 @@
...
-app.use(
-    session({
-        secret: "authorshaven",
-        cookie: { maxAge: 60000 },
-        resave: false,
-        saveUninitialized: false
-    })
-);
+app.get('/', (req, res) => {
+  res.status(200).json({ message: `hello ${process.env.USER}` });
```
Comment (kipropbrian) [inline src/index.js]: No need for displaying your username. Maybe `barefoot-api`

File: src/index.js [hunk 3]
```diff
@@ -1,93 +1,59 @@
...
-const isProduction = process.env.NODE_ENV === "production";
-
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import errorhandler from 'errorhandler';
+import morgan from 'morgan';
 
 const app = express();
+const isProduction = process.env.NODE_ENV;
```
Comment (kipropbrian) [inline src/index.js]: Is there a .env file?

**Review by kipropbrian — COMMENTED** [review]
Left some comments

File: src/index.js (hunk 3 above)
Comment (descholar-ceo) [reply src/index.js]: @kipropbrian  this `production` environment is set in `start script` of `package.json` file
as it is shown on the screenshoot

Screenshot from 2020-03-02 11-12-58

**Review by Ugizwenayo-Divine — APPROVED** [review]

**Comment by kipropbrian** [comment]
We need a .env file that will hold all the variables. Could you add that.

**Pull request closed without merging**